infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationServerHandler.java

```diff
@@ -42,30 +47,37 @@ public class GRPCLogReplicationServerHandler extends LogReplicationChannelGrpc.L
      * Note: we cannot rely on the request ID, because for client streaming APIs this will change for each
      * message, despite being part of the same stream.
      */
-    Map<Pair<UuidMsg, Long>, StreamObserver<ResponseMsg>> replicationStreamObserverMap;
+    Map<Pair<LogReplicationSession, Long>, StreamObserver<ResponseMsg>> replicationStreamObserverMap;
+
+    /*
+     * Map of session to StreamObserver to send requests to the clients.
+     */
+    Map<LogReplicationSession, StreamObserver<RequestMsg>> sessionToStreamObserverRequestMap;
 
-    public GRPCLogReplicationServerHandler(LogReplicationServerRouter router) {
+    public GRPCLogReplicationServerHandler(LogReplicationClientServerRouter router) {
         this.router = router;
         this.streamObserverMap = new ConcurrentHashMap<>();
         this.replicationStreamObserverMap = new ConcurrentHashMap<>();
+        this.sessionToStreamObserverRequestMap = new ConcurrentHashMap<>();
     }
 
     @Override
     public void negotiate(RequestMsg request, StreamObserver<ResponseMsg> responseObserver) {
-        log.trace("Received[{}]: {}", request.getHeader().getRequestId(),
+        log.info("Received[{}]: {}", request.getHeader().getRequestId(),
```

Comment:
hopefully you plan to convert this back to trace ? Is it hard to debug with trace on ?

Reply:
Yup, this will be reverted once we resolve the flaky tests